examples/Dockerfile.service

```diff
@@ -0,0 +1,64 @@
+# Define the Container registry for our base images, as these are not stored on Dockerhub.
+#   You will need to be logged into the registry in order to build this.
+ARG REGISTRY=562955126301.dkr.ecr.eu-west-2.amazonaws.com
+ARG BASE_VERSION=1.0.0
+ARG DEV_VERSION=1.0.0
+FROM $REGISTRY/ffc-node-base:$BASE_VERSION
+
+# Make sure that we don't run as root. The "node" user is set up in the base node version that our base parent image uses.
+USER node
+
+WORKDIR /home/node
+
+# Copy our package files so that our package install will do a clean install. This installs the exact versions of the packages
+#   listed in package-lock.json, and does not update either the package-lock.json or the package.json file.
+# Our production dependancies are now installed.
+COPY --chown=node:node package*.json ./
+RUN npm ci
+
+# Development stage installs devDependencies, builds app from source and declares a file watcher as the default command.
+#   We name this stage so we can refer to it in later stages
+FROM $REGISTRY/ffc-node-development:$DEV_VERSION AS development
+
+# Expose some development debugging ports that are used during development
+EXPOSE 9229 9230
+
+# As this is a new container, we want to ensure that we set up the working directory and user correctly
+USER node
+WORKDIR /home/node
+
+# We copy the installed packages from the production "base" install that was the first stage.
+COPY --from=base --chown=node:node /home/node/package*.json ./
+COPY --from=base --chown=node:node /home/node/node_modules ./node_modules
+
+# We run a full dev install to bring in any development packages
+RUN npm install
+COPY --chown=node:node app/ ./app/
+
+# We specify a command here, as we can run this stage directly using either docker-compose files, or passing --target to the docker build command
+CMD [ "npm", "run", "start:watch" ]
+
+# Test stage copied in Jest configuration and declares the test task as the default command. We use the development stage
+#   for this as it will have all the tools required installed and have everything ready to run tests
+FROM development AS test

```

Comment:
does there need to be a separate test stage? The development stage already includes all the test framework files via the npm development packages, so copying the test files across in the development stage would avoid this stage completely, as the test command is specified in the docker compose test file.

To my mind there's only two flavours of docker files for a service - a development build and a production build.

Reply:
The test stage copies over further test files, I think it makes sense but it's something that can be easily removed later if we decide that it's not needed.